### PR TITLE
don't generate default routes when generating plugin api

### LINKED ISSUE #3965: Scaffolding APIs on plugin seems broken
**Describe the bug**
A simple scaffolding and creating a api on a plugin seems broken.

**Steps to reproduce the behavior**
```bash
$ strapi new myapp
$ cd myapp/

$ strapi generate:plugin myplugin
$ strapi generate:api -p myplugin mymodel first:string second:string

$ yarn develop

$ strapi develop
[2019-09-11T14:52:25.574Z] debug ⛔️ Server wasn't able to start properly.
[2019-09-11T14:52:25.579Z] error TypeError: Cannot read property 'bind' of undefined
    at routerChecker (/Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/router/utils/routerChecker.js:33:43)
    at /Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/router/utils/composeEndpoint.js:16:34
    at _.forEach.value (/Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/router/index.js:63:15)
    at /Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:4905:15
    at baseForOwn (/Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:2990:24)
    at /Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:4874:18
    at Function.forEach (/Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:9342:14)
    at _.forEach (/Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/router/index.js:60:13)
    at /Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:4905:15
    at baseForOwn (/Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:2990:24)
    at /Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:4874:18
    at Function.forEach (/Users/ollebroms/Documents/testing/myapp/node_modules/lodash/lodash.js:9342:14)
    at Object.initialize (/Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/router/index.js:49:11)
    at Promise.resolve.then (/Users/ollebroms/Documents/testing/myapp/node_modules/strapi/lib/middlewares/index.js:54:28)
    at process._tickCallback (internal/process/next_tick.js:68:7)
error Command failed with exit code 1.
```

**Expected behavior**
I except the application to start and it should show the model in the sidebar and I don't except that I need to provide any controller actions.

This will work if it's not inside a plugin and I don't have to provide controller actions.
```bash
$ strapi new myapp
$ cd myapp/

$ strapi generate:api mymodel first:string second:string

$ yarn develop
```
It will automatically show up in the sidebar:
image

**System**
- Node.js version: v10.16.3
- NPM version: 6.9.0
- Strapi version: 3.0.0-beta.16.1
- Database: any
- Operating system: any

### PR DESCRIPTION
**Pull request by petersg83**
fix #3965

### CODE REVIEW

File: packages/strapi-generate-api/json/routes.json.js [hunk 1]
```diff
@@ -97,46 +97,20 @@ function generateCollectionTypeRoutes({ route, name }) {
...
-        'routes',
-        _.concat(routes, _.differenceWith(current.routes, routes, _.isEqual))
-      );
-
-      return current;
-    } catch (e) {
-      console.error(e);
-      return;
-    }
+    console.log('pouet');
```
Comment (alexandrebodin) [inline packages/strapi-generate-api/json/routes.json.js]: I'm pretty sure users would like this log but still :D

**Comment by derrickmehaffy** [comment]
This pull request has been mentioned on **Strapi Community**. There might be relevant details there:

https://forum.strapi.io/t/supporting-modular-programming-via-plugin-system/1090/9

File: packages/strapi-generate-api/json/routes.json.js [hunk 2]
```diff
@@ -97,46 +97,20 @@ function generateCollectionTypeRoutes({ route, name }) {
...
-      ? generateSingleTypeRoutes({
-          route: scope.route,
-          name: scope.name,
-        })
-      : generateCollectionTypeRoutes({
-          route: scope.route,
-          name: scope.name,
-        });
+  let routes = [];
+  if (!scope.args.plugin) {
```
Comment (sunnysonx) [inline packages/strapi-generate-api/json/routes.json.js:101]: Hello @alexandrebodin & @petersg83, What's the reason for disabling the API routes generation inside plugins? I've deleted that `if (!scope.args.plugin) {` locally and it seems to generate/merge existing routes ok.

Comment (alexandrebodin) [reply packages/strapi-generate-api/json/routes.json.js:101]: As the plugin models don't get auto generated controllers those default routes point to nothing. There was no reason to generate them :)

Comment (sunnysonx) [reply packages/strapi-generate-api/json/routes.json.js:101]: In that case, it also brokes the functionality of the `strapi generate:api order --plugin shop` command,  which should create a new API inside the plugin, it doesn't generate the routes for it automatically.

Comment (sunnysonx) [reply packages/strapi-generate-api/json/routes.json.js:101]: There should be added an extra condition on that `if`, that verifies if the `generate:model` was used.

**Comment by derrickmehaffy** [comment]
This pull request has been mentioned on **Strapi Community**. There might be relevant details there:

https://forum.strapi.io/t/supporting-modular-programming-via-plugin-system/1090/11

**Pull request merged**